### PR TITLE
refactor(ngcc): improve logging of progress

### LINKED ISSUE #42308: Show warnings when depending upon a non-Ivy library in v13+
# 🚀 Feature request

### Command (mark with an `x`)

- [x] new
- [x] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description

Show a warning for each non-Ivy library to encourage people to move their libraries to Ivy.

For example, if an Angular project depends on a non-Ivy library in ngcc we can show a warning:

```
The library XYZ is distributed in a non-Ivy format. Encourage the library author to publish an Ivy distribution by opening a feature request at [package-json.repo].
```

### PR DESCRIPTION
**Pull request by petebacondarwin**
This commit adds additional information to encourage developers to contact
the author of a View Engine library and ask them to update to partial Ivy.

Fixes #42308

### CODE REVIEW

File: packages/compiler-cli/ngcc/src/execution/create_compile_function.ts [hunk 1]
```diff
@@ -55,7 +55,8 @@ export function getCreateCompileFn(
         return;
       }
 
-      logger.info(`Compiling ${entryPoint.name} : ${formatProperty} as ${format}`);
+      logger.info(`- ${entryPoint.name} (${entryPoint.repositoryUrl})`);
+      logger.debug(`    property: ${formatProperty}, format: ${format}`);
```
Comment (JoostK) [inline packages/compiler-cli/ngcc/src/execution/create_compile_function.ts:59]: My vote would be to keep the format in the info message. Especially for command-line invocations it's a bit more typical that more than one format is being processed and without the format we'll see identical messages for the same entry-point.

File: yarn.lock [hunk 2]
```diff
@@ -14916,7 +14914,7 @@ write-file-atomic@^3.0.0:
     signal-exit "^3.0.2"
     typedarray-to-buffer "^3.1.5"
 
-ws@7.4.6, ws@~7.4.2:
+ws@7.4.6:
```
Comment (JoostK) [inline yarn.lock:14917]: The lockfile changes appear unrelated to the changes so you may want to revert.

Comment (petebacondarwin) [reply yarn.lock:14917]: Oh yes - not sure how this happened.
Confused why renovate did not resolve this previously....

File: packages/compiler-cli/ngcc/src/execution/create_compile_function.ts (hunk 1 above)
Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/execution/create_compile_function.ts:59]: I am not convinced that multiformat runs of ngcc are more typical even on the command line.
Obviously the CLI will target only one property per entry-point.

But command line users who are not doing something similar (by providing a list of acceptable properties and asking to only build the first one) are probably wasting a lot of time processing formats that are not used.

But I think it is reasonable to keep one of the two (property or format) in the info message. But I found that keeping both just added a lot of noise to the message, making it less helpful.

Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/execution/create_compile_function.ts:59]: which do you think is most helpful the prop or the format?

Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/execution/create_compile_function.ts:59]: I went with the property, since this is more fine-grained that the format and you can compute the format from the property but not necessarily the other way round.

**Review by JoostK — APPROVED** [review]

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by petebacondarwin** [comment]
Caretaker: The saucelabs failures are not related to this PR.

**Comment by atscott** [comment]
This PR was merged into the repository by commit 28ef5af03fbeb371f1ecf191ca629815f40e5a22.
